lib/ElementHandle.js

```diff
@@ -103,6 +103,26 @@ class ElementHandle {
     const objectId = this._remoteObject.objectId;
     return this._client.send('DOM.setFileInputFiles', { objectId, files });
   }
+
+  /**
+   * @return {!Promise<Object>}
+   */
+  async boundingBox() {
+    const box = await this.evaluate(element => {
+      if (!element.ownerDocument.contains(element))
+        return null;
+      const rect = element.getBoundingClientRect();
```

Comment:
This returns bounding box in viewport, whereas `screenshot` accepts coordinates relative to the main frame. To make this usable with `page.screenshot`, you need to convert coordinates to the window.

That's how we do this in DevTools front-end: [DOMExtension.js](https://cs.chromium.org/chromium/src/third_party/WebKit/Source/devtools/front_end/dom_extension/DOMExtension.js?type=cs&q=AnchorBox+f:front_end&sq=package:chromium&l=494)

Let's add a test that verifies screenshots with clipping, something along the lines:
```js
describe('Page.screenshot', function() {
  ...
  it('should work with elementHandle.boundingBox', SX(async function() {
    await page.setViewport({width: 500, height: 500});
    await page.goto(PREFIX + '/grid.html');
    await page.evaluate(() => window.scrollBy(0, 100));
    const box = await page.$('div:nth-child(3)');
    const screenshot = await page.screenshot({clip: await box.boundingBox()});
    expect(screenshot).toBeGolden('screenshot-element-bounding-box.png');
  }));
  ...
```